signer/src/testing/storage/model.rs

```diff
@@ -323,7 +318,7 @@ impl TestData {
         num_stacks_blocks: usize,
     ) -> Vec<model::StacksBlock> {
         let mut stacks_block: model::StacksBlock = fake::Faker.fake_with_rng(rng);
-        stacks_block.bitcoin_anchor = new_bitcoin_block.parent_hash;
+        stacks_block.bitcoin_anchor = new_bitcoin_block.block_hash;
```

Comment:
Originally this was the parent hash as the stacks block that will be "included" in `new_bitcoin_block` are actually anchored to its parent. Now, anchoring to the current block, we are in some sense offsetting the stacks chain wrt the bitcoin one. Why do we want this?

Reply:
The nice thing about doing it this way is that each Stacks block has an anchor block that is also in the database. The current code had the issue where the first Stacks block(s) had no known anchor block. So I think this fixes one of our tests (https://github.com/stacks-network/sbtc/issues/1466).

Besides that small benefit, I couldn't think of any downsides.